### PR TITLE
[done][needs test merge][waiting for test merge since 3 weeks ago] curb your roundend 2: curb harder: return of curb

### LINKED ISSUE #20251: Maybe revert "new" roundend sounds
Discussion rather than a demand
By "new" roundend sounds I mean all the ones that are not "banging donk" or "apc destroyed"
We've had these for 2 years and a half so that's plenty of time to get used to them, and still I find that the reception to them is not very positive
https://www.strawpoll.me/16659929/r
firefox_2018-10-16_23-28-55

so maybe we can have a formal discussion or a server poll on them

Additionally the bulk of them were merged in less than 16 hours #9964 which I'm still salty about

### PR DESCRIPTION
**Pull request by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE**
Revival of #18207, but with one very big difference:
These credits play client-side WHILE THE SERVER REBOOTS and only start playing 1 second before the server shuts down - no roundend grief interruption
So, in theory, these would replace the roundend sounds (fixes #20251)

New And Updated Webm: http://webmshare.com/YnZna

WHY THIS IS A WIP:
 - [x] ~~I need to test this in the live serb to see if the sound keeps playing correctly while the server resets. If it doesn't I'll have to change it a bit.~~ seemed to work acceptably with 1 player online, pending live serb test merge

WHAT NEEDS FINISHING:
 - [x] ~~None of the formatting is done I literally just salvaged the text from celt's PR and is working with absolutely no formatting yet. I also need to create actually working CSS so the credits are narrower and the scrolling text starts off-screen.~~ done
 - [x] ~~For some reason the credits are not correctly clearing upon re-login if the player doesn't close the game during reboot - worked around, albeit shittily~~ done
 - [x] ~~Test what happens when an admin delays roundend (what should happen?)~~ apparently admins just have to manually use the "reboot" verb when round end is delayed, correct me otherwise
 - [x] ~~Test what happens when an admin forces reboot (what should happen?)~~ credits roll correctly when forcing reboot unless it's a high priority reboot

:cl:
 * rscadd: round end now has a credits sequence inspired by Curb Your Enthusiasm

### CODE REVIEW

File: code/modules/credits/credits.dm [hunk 1]
```diff
@@ -0,0 +1,137 @@
+/mob/verb/creditstest()
```
Comment (I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE) [inline code/modules/credits/credits.dm]: HITLERS

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
By the way I loathe CSS on BYOND because any of the things people did after 2009 don't work in it so if someone can hook me up with a CSS layout that goes like this using only pre-2009 technology
photoshop_2018-10-20_04-55-42
I'd love that

**Comment by Riekland** [comment]
>So, in theory, these would replace the roundend sounds 

NO

don't you touch yeah apc destroyed and donk

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
They died 2.5 years ago when they were made to play 10% of the time

**Comment by Riekland** [comment]
you know that's not an arguement you sonnovabitch how dare you make an issue to remove all the shit ones then do this I trusted you

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
Holy fuck I'm so mad I spent TWO HOURS recompiling and retesting because of some nasty mysterious bug and it turned out to be javascript fffffffffuuuuuuuuuuuuuuuuuuuuck anyways later tonight I'll add a webm of the finished formatting

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
About banging donk and client preferences to choose old sounds  instead of credits etc:

There are two ways I could play the roundend sounds

One's what the server is already doing. It works and we know it works. However in order to not be interrupted by server shutdown the same sound must be sent to everyone, no room for client prefs. Also, the live server is doing something weird because this doesn't act work for me in a listen server, I'm just assuming it will work when testmerged.

Number two would be sending the .ogg via asset and playing it in a client side browser player. I'm 100% sure this would keep playing after disconnection and it would allow for client preferences.
However, I don't want to do this, because it's a huge back and there's probably 10 yet to be seen ways it will break or get me yelled at.

So, I want to test method 1 in the test server, and if it does inexplicably work, well, char prefs be damned and emojicracy will decide. If it doesn't work, I'll have to hack together method #2 and might as well add char prefs there.

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
Done, now it just needs test server testing (preferably by 2-3+ people) to see if the sound gets interrupted or not

Also, can someone with experience tell me if the .ogg should be shortened or not? I know admemes often play .oggs twice the size but there might be concerns with playing this so close to the server closure.

**Comment by SGuivenne** [comment]
Serious question, if you can send full sound files, couldn't you use byond to straight up send malware at server end?  This seems incredibly exploitable.

**Comment by Exxion** [comment]
1. No
2. Even if you could, it would be the case regardless of whether or not this was added
3. What kind of toaster are you playing on that can be crashed by a MIDI?

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
can I get a uhhhhhhh te:b:t ser:b:er merge

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
Taking ideas for episode names/disclaimers

File: code/controllers/configuration.dm [hunk 2]
```diff
@@ -166,6 +166,7 @@
 	var/mommi_static = 0 //Scrambling mobs for mommis or not
 
 	var/skip_minimap_generation = 0 //If 1, don't generate minimaps
+	var/skip_holominimap_generation = 0 //If 1, don't generate holominimaps
```
Comment (DamianX) [inline code/controllers/configuration.dm:169]: did you intend to put this in this PR?

Comment (I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE) [reply code/controllers/configuration.dm]: No I committed it by accident
but I don't really want to revert it and move it to another PR now

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
Please put this in the test server.

**Comment by Exxion** [comment]
Done

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
Not only this isn't in the test server but the test server seems to be running 2 months old code in general

**Comment by Exxion** [comment]
Yeah it always is
But what the fuck it should be on there

**Comment by Exxion** [comment]
Yeah it's on the branch
Did you try restarting it?

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
yeah I'm sure

**Comment by SonixApache** [comment]
test serb is up

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
Okay, as predicted the test server, just like the main server, does something very weird that I can't explain and the sound either does not interrupt or is much harder to interrupt post-disconnection there, unlike a listen server.

Question remains whether it'll work for everyone and if the fact that there might be more than 1 player online at the same time might lead to wonkiness regardless.
I'm absolutely not going to be able to get 10 people to join the test server and test this "properly" so please test merge this and if it doesn't work right then just revert it after a couple rounds after we get usable bug reports.

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
@DamianX @Exxion @Intigracy @MadmanMartian

**Comment by MadmanMartian** [comment]
ree

**Comment by jknpj** [comment]
he wants a LIVE test though

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
@DamianX @Exxion @Intigracy @MadmanMartian @jknpj 
Please take a look at this PR, it's been ready for a while and just waiting for a collab

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
Thanks. There's a chance the sound at the end won't work correctly (I really can't test this on my end) so if it's way too broken you have my blessing to revert it and I'll open a fixed PR later.

**Comment by Intigracy** [comment]
the entire thing just straight up isn't working on live

**Comment by SonixApache** [comment]
smelelel -> griff -> bangingdonk
no curb whatsoever

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
I removed banging donk completely so if that soundbyte played it sounds like the server's not pulling new changes again
I'll try to observe next roundend

**Comment by Intigracy** [comment]
The skip holomap generation config option is present on live, so it's merged.

Still just heard smelelel.

**Comment by Intigracy** [comment]
Spoke too soon, I heard smelelel and then saw the roundend credits.

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
Worked for me, the sound interrupted after about 30 seconds which is not great

**Comment by DamianX** [comment]
The sound was playing twice for me.

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
2 times on top of eachother or it started playing again later?

**Comment by DamianX** [comment]
On top of each other, delayed by a few seconds.

File: code/controllers/subsystem/init/more_init_stuff.dm [hunk 3]
```diff
@@ -20,10 +20,15 @@ var/datum/subsystem/more_init/SSmore_init
...
-	log_startup_progress("  Finished holominimaps in [stop_watch(watch)]s.")
+	if (!config.skip_minimap_generation)
+		watch=start_watch()
+		log_startup_progress("Generating holominimaps...")
+		generateHoloMinimaps()
+		log_startup_progress("  Finished holominimaps in [stop_watch(watch)]s.")
+	else
+		holomaps_initialized = 1 //Assume holominimaps were prerendered, the worst thing that happens if they're missing is that the minimap consoles don't show a minimap
+		log_startup_progress("Not generating holominimaps - SKIP_HOLOMINIMAP_GENERATION found in config/config.txt")
+	..()
```
Comment (DeityLink) [inline code/controllers/subsystem/init/more_init_stuff.dm:31]: why

Comment (Exxion) [reply code/controllers/subsystem/init/more_init_stuff.dm:31]: I'm sure having two supercalls in there breaks something, but I'm curious what

Comment (DamianX) [reply code/controllers/subsystem/init/more_init_stuff.dm:31]:
```
/datum/subsystem/proc/Initialize(start_timeofday)
	var/time = (world.timeofday - start_timeofday) / 10
	var/msg = "Initialized [name] subsystem within [time] seconds!"
	log_debug(msg,FALSE) // Redundant chat output as whole world sees message below anyways
	world.log << msg
	to_chat(world, "<span class='danger'>[msg]</span>")
	initialized = TRUE
	return time
```
It's fine.

Comment (Exxion) [reply code/controllers/subsystem/init/more_init_stuff.dm:31]: Well the message is duplicated, but that's not really a big deal

Comment (DamianX) [reply code/controllers/subsystem/init/more_init_stuff.dm:31]: I thought that was the intent.

Comment (Exxion) [reply code/controllers/subsystem/init/more_init_stuff.dm:31]: But for what purpose?

Comment (DeityLink) [reply code/controllers/subsystem/init/more_init_stuff.dm:31]: I was under the impression that it was a typo

**Pull request merged**